src/option.h

```diff
@@ -616,6 +616,46 @@ namespace librealsense
        std::string _reason;
    };
 
+   class multi_gated_option : public proxy_option
```

Comment:
What would be the effort to refactor the base class to do this ?

Reply:
Done